srv/salt/_modules/dg.py

```diff
@@ -534,6 +508,17 @@ def compare(self, disk: dict) -> bool:
         return False
 
 
+class Inventory(object):
+    def __init__(self, include_unavailable=False):
+        self.include_unavailable = include_unavailable
+
+    @property
+    def disks(self) -> list:
+        """ Returns a list of disks (json_report)"""
+        return __salt__['cephdisks.all'](kwarg=dict(
+            exclude_available=self.include_unavailable))
```

Comment:
Does `exclude_available=self.include_unavailable` do the right thing here?

Reply:
I can't follow this logic anymore either. I hate it and it sucks, but for me it seems to do what I expect.

Reply:
This is still bothering me. Looking at the names, if I want to exclude available disks, that doesn't imply anything about whether or not I want to include or exclude _unavailable_ disks. ISTM they're two separate conditions.

Reply:
I guess I have to rework this..

Reply:
Sorry :-(

Reply:
well, I was kinda procrastinating on that :> but it's probably a really bad sign if I can't explain the logic I wrote a month ago :wave:

Reply:
So, I overhauled it and got rid of all `'include_unavailable<->exclude_available'` mappings.

Which flags are used is now determined by the method used.

- deploy 

Here you want to use disks that are available, not used by ceph -> **unused**

- c_v_commands

Same as deploy, as deply calls c_v_commands **unused**

- list/report

This generates now a annotated list of all devices, hence it also needs **all** devices (unavailable, used_by_ceph - no root_device though)


`used/unused/all` are offloaded to `cephdisks.py` and are also available as standalone/callable functions.

`cephdisks.unused/used/all`